### PR TITLE
feat(user-service) implement user retrieval and registration service

### PR DESCRIPTION
**Pull request by will0684**
This pull request introduces basic functionality for managing users, including backend service enhancements, frontend service implementations, and mock data for testing. The changes enable retrieving and registering users, both in the backend and frontend, while adding robust error handling and comprehensive tests.

### Backend Enhancements:
* Added `Optional<UserEntity> getUserById(Long id)` method to `UserService` for retrieving users by ID.
* Introduced a new endpoint in `UsersController` to fetch user details by ID, with proper security annotations and error handling.

### Frontend Service Implementations:
* Defined `User` and `UserCreate` types in `models.ts` for user data structures.
* Created `user-service-default.ts` to interact with the backend API for retrieving and registering users, with detailed error handling.
* Added `user-service-mock.ts` for mock implementations of user retrieval and registration, supporting testing and development environments.
* Implemented `user-service.ts` to dynamically switch between default and mock services based on environment configuration.

### Testing:
* Added unit tests in `user-service-mock.test.ts` to verify the functionality of mock user service methods, including edge cases for user retrieval and registration.

### CODE REVIEW

File: backend/src/main/java/ca/gov/dtsstn/vacman/api/web/UsersController.java [hunk 1]
```diff
@@ -57,4 +58,12 @@ public UserReadModel createUser(@RequestBody @Valid UserCreateModel user) {
 		return userModelMapper.toModel(userService.createUser(userModelMapper.toEntity(user)));
 	}
 
+	@GetMapping("/{id}")
+	@Operation(summary = "Get a user by ID.")
+	@SecurityRequirement(name = SpringDocConfig.AZURE_AD)
+	public UserReadModel getUserById(@PathVariable Long id) {
+		return userService.getUserById(id)
+			.map(userModelMapper::toModel)
+			.orElseThrow(() -> new RuntimeException("User not found"));
```
Comment (gregory-j-baker) [inline backend/src/main/java/ca/gov/dtsstn/vacman/api/web/UsersController.java]: This is okay for now, but it will have to be changed. Typically what I do here is create a new `ResourceNotFoundException` that I throw here.. and then configure spring to return a 404 status when it encounters that.

For reference: <https://github.com/DTS-STN/passport-status-api/blob/main/src/main/java/ca/gov/dtsstn/passport/api/web/ApiErrorHandler.java#L208>

The passport response object is NIEMified, so we can't just copy/paste it.. but the same general approach should be used in this API.

File: frontend/tests/.server/domain/services/user-service-mock.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      expect(createdUser).toMatchObject({
+        id: '4', // Should be next ID after the 3 mock users
+        name: 'Test User',
+        createdBy: 'system',
+        lastModifiedBy: 'system',
+      });
+
+      // Check that dates are ISO strings (exact time will vary)
+      expect(createdUser.createdDate).toMatch(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/);
+      expect(createdUser.lastModifiedDate).toMatch(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/);
```
Comment (gregory-j-baker) [inline frontend/tests/.server/domain/services/user-service-mock.test.ts:47]: You *could* use `isValidDateString(..)` from `date-utils.ts` here if you wanted.. although you could argue that doing so would make this a non-unit test.

Alternatively you could just do exactly what that function is doing: `isValid(parseISO(date))`

File: backend/src/main/java/ca/gov/dtsstn/vacman/api/web/UsersController.java (hunk 1 above)
Comment (gregory-j-baker) [reply backend/src/main/java/ca/gov/dtsstn/vacman/api/web/UsersController.java]: Yep.. I'll resolve this. See my other comment: <https://github.com/DTS-STN/vacman/pull/89#discussion_r2126625941>

**Review by gregory-j-baker — APPROVED** [review]
